### PR TITLE
Upgrade copyTree

### PR DESCRIPTION
**Pull request by kriskowal**
To support target file system and include testing.

### CODE REVIEW

**Comment by Stuk** [comment]
On mobile at the moment so can't comment on the line. 

`return sourceFs.copy(source, target);` looks wrong. Shouldn't there be the targetFs?

**Comment by kriskowal** [comment]
Good catch.

**Comment by kriskowal** [comment]
Also, pending tests and a corresponding change to v2. Not this week though.

File: fs-common.js [hunk 1]
```diff
@@ -157,34 +157,54 @@ exports.update = function (exports, workingDirectory) {
...
+        targetFs = options.targetFs || sourceFs;
+        includeTest = options.include || returnTrue;
+        return sourceFs.statLink(source).then(function (stat) {
+            var include = includeTest(source, stat);
+            return Q(include).then(function (include) {
+                if (!include) return;
+                if (stat.isFile()) {
+                    return sourceFs.copy(source, target);
+                } else if (stat.isDirectory()) {
+                    return sourceFs.exists(target).then(function (targetExists) {
```
Comment (Stuk) [inline fs-common.js:172]: Should be `targetFs.exists`

**Comment by kriskowal** [comment]
Fixes #113

**Comment by hthetiot** [comment]
Why dont we merge @kriskowal ?

File: fs-common.js (hunk 1 above)
Comment (hthetiot) [reply fs-common.js:172]: ACK

**Pull request closed without merging**